components/core/src/clp/GlobalMetadataDBConfig.hpp

```diff
@@ -16,11 +18,36 @@ class GlobalMetadataDBConfig {
         MySQL,
     };
 
+    /**
+     * Overloads operator >> to read MetadataDBType from an input stream.
+     * @param in
+     * @param metadata_db_type
+     * @return std::istream& reference to the input stream
+     */
+    friend auto operator>>(std::istream& in, MetadataDBType& metadata_db_type) -> std::istream&;
```

Comment:
Just to make sure I understand, this will work with argument parsing code?

```
("db-type",
             boost::program_options::value<MetadataDBType>(&m_metadata_db_type)->default_value(cDefaultMetadataDbType, cDefaultMetadataDbTypeName.data()),